### ISSUE TITLE
AnimationController.reverse() called after AnimationController.dispose()

### ISSUE THREAD

**Issue by shemhazai** [ponnamkarthik/FlutterToast#222]
## Steps to reproduce:
- Show a custom toast with FToast(context).showToast(...);
- Dismiss the toast by calling FToast(context).removeCustomToast(); before the toastDuration passes
- An exception is thrown.

## Proposal:
Do not create unmanaged futures like in initState(), create a timer which gets cancelled when a toast is removed.

Mentioned code which needs fixes:
```
  @override
  void initState() {
    _animationController = AnimationController(
      vsync: this,
      duration: const Duration(milliseconds: 350),
    );
    _fadeAnimation = CurvedAnimation(parent: _animationController, curve: Curves.easeIn);
    super.initState();

    showIt();

    Future.delayed(widget.duration, () {
      hideIt();
    });
  }
```

## Stacktrace:
```
I/flutter ( 5710): 2020-08-30 18:08:32.448241/ 'package:flutter/src/animation/animation_controller.dart': Failed assertion: line 490 pos 7: '_ticker != null': AnimationController.reverse() called after AnimationController.dispose()
I/flutter ( 5710): AnimationController methods should not be used after calling dispose.
I/flutter ( 5710): 2020-08-30 18:08:32.448626/ #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:46:39)
I/flutter ( 5710): #1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:36:5)
I/flutter ( 5710): #2      AnimationController.reverse (package:flutter/src/animation/animation_controller.dart:490:7)
I/flutter ( 5710): #3      ToastStateFulState.hideIt (package:fluttertoast/fluttertoast.dart:192:26)
I/flutter ( 5710): #4      ToastStateFulState.initState.<anonymous closure> (package:fluttertoast/fluttertoast.dart:210:7)
I/flutter ( 5710): #5      new Future.delayed.<anonymous closure> (dart:async/future.dart:325:39)
I/flutter ( 5710): #6      _rootRun (dart:async/zone.dart:1182:47)
I/flutter ( 5710): #7      _CustomZone.run (dart:async/zone.dart:1093:19)
I/flutter ( 5710): #8      _CustomZone.runGuarded (dart:async/zone.dart:997:7)
I/flutter ( 5710): #9      _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
I/flutter ( 5710): #10     _rootRun (dart:async/zone.dart:1190:13)
I/flutter ( 5710): #11     _CustomZone.run (dart:async/zone.dart:1093:19)
I/flutter ( 5710): #12     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1021:23)
I/flutter ( 5710): #13     Timer._createTimer.<anonymou

```

**Comment by xJon** [comment]
Can confirm this issue as well.

**Comment by ponnamkarthik** [comment]
Fixed in `7.1.0`
